kbcstorage/files.py

```diff
@@ -193,32 +178,121 @@ def download(self, file_id, local_path):
             os.mkdir(local_path)
         file_info = self.detail(file_id=file_id, federation_token=True)
         local_file = os.path.join(local_path, file_info['name'])
-        s3 = boto3.resource(
-            's3',
-            aws_access_key_id=file_info['credentials']['AccessKeyId'],
-            aws_secret_access_key=file_info['credentials']['SecretAccessKey'],
-            aws_session_token=file_info['credentials']['SessionToken'],
-            region_name=file_info['region']
-        )
-        if file_info['isSliced']:
-            manifest = requests.get(url=file_info['url']).json()
-            file_names = []
-            for entry in manifest["entries"]:
-                full_path = entry["url"]
-                file_name = full_path.rsplit("/", 1)[1]
-                file_names.append(file_name)
-                splitted_path = full_path.split("/")
-                file_key = "/".join(splitted_path[3:])
-                bucket = s3.Bucket(file_info['s3Path']['bucket'])
-                bucket.download_file(file_key, file_name)
-            # merge the downloaded files
-            with open(local_file, mode='wb') as out_file:
-                for file_name in file_names:
-                    with open(file_name, mode='rb') as in_file:
-                        for line in in_file:
-                            out_file.write(line)
-                    os.remove(file_name)
-        else:
-            bucket = s3.Bucket(file_info["s3Path"]["bucket"])
-            bucket.download_file(file_info["s3Path"]["key"], local_file)
+        if file_info['provider'] == 'azure':
```

Comment:
Similarly to @odinuv comment in tests - what if the provider is for any reason something else than `azure` or `aws`?

Reply:
ok, I changed it to assert that provider before this switch